var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -149,11 +153,17 @@ def edit(self, spec, prefix):
         # ESMF_COMM must be set to indicate which MPI implementation
         # is used to build the ESMF library.
         if '+mpi' in spec:
-            if '^mvapich2' in spec:
+            if 'platform=cray' in self.spec:
+                os.environ['ESMF_COMM'] = 'mpi'
+            elif '^mvapich2' in spec:
                 os.environ['ESMF_COMM'] = 'mvapich2'
             elif '^mpich' in spec:
-                # FIXME: mpich or mpich2?
+                # esmf@7.0.1 does not include configs for mpich3,
+                # so we start with the configs for mpich2:
```

Comment:
There is no such thing as `mpich3`. The original project was named `mpich`, then version 2 rolled out and they renamed it to `mpich2`, then when 3 came along they decided to change the name back to `mpich`. So I'm not sure if `mpich` refers to `mpich1` or `mpich3` or both... But if it builds, that's a good sign. At least, that was my understanding when I wrote this package.